### PR TITLE
Add/e2e implement describe each

### LINKED ISSUE #29678: Implement describe.each where appropriate
There are several tests that repeat the same test steps with different data (eg. order search). We should update these tests to use `describe.each()` to reduce the code size and ensure consistency among each set of tests.

### PR DESCRIPTION
**Pull request by rodelgc**
### Changes proposed in this Pull Request:

- This PR implements `describe.each` and `it.each` on tests that have repetitive steps but use different test data on each iteration. This will not improve performance that much, but will make our tests more concise and readable.
- It contains necessary cherry-picked commits from #30184.
- I refactored [this variable](https://github.com/woocommerce/woocommerce/pull/30119/files#diff-4c0e5f5a585ade4c9bf11d9c57b30abf211cbf3be265168d6017ebd7b7945cd3L15-R23) to make it easier to access its values from the `each`'ed test.

Closes #29678.

### How to test the changes in this Pull Request:

1. Set up e2e environment by running the onboarding & activation tests:
    ```
    npx wc-e2e test:e2e ./tests/e2e/specs/activate-and-setup/setup-onboarding.js
    ```
2. Run the "merchant order search" test:
    ```
    npx wc-e2e test:e2e-dev ./tests/e2e/specs/wp-admin/order-searching.test.js
    ```
3. All these tests should pass:
    Screen Shot 2021-06-17 at 3 45 14 PM
4. Run the other tests where we implemented `each()`:
    1. `E2E_RETEST=1 npx wc-e2e test:e2e ./tests/e2e/specs/activate-and-setup/onboarding-tasklist.test.js`
        Screen Shot 2021-07-01 at 2 24 20 PM
    1. `npx wc-e2e test:e2e ./tests/e2e/specs/wp-admin/analytics-page-loads.test.js`
        Screen Shot 2021-07-01 at 3 29 28 PM
    1. `npx wc-e2e test:e2e ./tests/e2e/specs/wp-admin/order-status-filters.test.js`
        Screen Shot 2021-07-02 at 6 53 25 PM
    1. `npx wc-e2e test:e2e ./tests/e2e/specs/front-end/cart-coupons.test.js`
        Screen Shot 2021-07-01 at 7 22 34 PM
    1. `npx wc-e2e test:e2e ./tests/e2e/specs/front-end/checkout-coupons.test.js`
        Screen Shot 2021-07-01 at 7 32 31 PM
    1. `npx wc-e2e test:e2e ./tests/e2e/specs/front-end/my-account.test.js`
        Screen Shot 2021-07-01 at 7 49 20 PM

        

### Other information:
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Implemented data-driven design using `describe.each` and `it.each`

### CODE REVIEW

File: tests/e2e/core-tests/specs/merchant/wp-admin-order-status-filters.test.js [hunk 1]
```diff
@@ -4,178 +4,71 @@
...
-};
+const orderStatus = [
+	['Pending payment', 'wc-pending'],
+	['Processing', 'wc-processing'],
+	['On hold', 'wc-on-hold'],
+	['Completed', 'wc-completed'],
+	['Cancelled', 'wc-cancelled'],
+	['Refunded', 'wc-refunded'],
+	['Failed', 'wc-failed']
+];
```
Comment (rodelgc) [inline tests/e2e/core-tests/specs/merchant/wp-admin-order-status-filters.test.js:23]: I had to refactor this variable so it's easier to access its values from the `each`'ed test.

**Comment by rodelgc** [comment]
I just saw the merge conflict above. Will sort it out today or tomorrow.

**Comment by rrennick** [comment]
@rodelgc I'll review this one after #30184 is merged to trunk and trunk has been merged to this branch.

**Comment by zhongruige** [comment]
@rodelgc looks like once we merged in https://github.com/woocommerce/woocommerce/pull/30184 there's some merge conflicts on this PR. Once we get those resolved and trunk merged in I'll take a look at reviewing this PR.

**Comment by rodelgc** [comment]
Conflicts resolved. Already synced with `trunk`. I cherry-picked the fix for the WCCOM issue by the way (the issue mentioned in p1630361173018400-slack-C01BWDDTGKX). Ready for another round of review.

File: tests/e2e/core-tests/specs/activate-and-setup/onboarding-tasklist.test.js [hunk 2]
```diff
@@ -35,14 +35,17 @@ const runOnboardingFlowTest = () => {
 				await withRestApi.deleteAllShippingZones();
 			});
 
-			it('can reset shipping classes', async () => {
-				await withRestApi.deleteAllShippingClasses();
-			})
+			it.each(['general', 'products', 'tax'])(
```
Comment (rrennick) [inline tests/e2e/core-tests/specs/activate-and-setup/onboarding-tasklist.test.js]: Kudos on noticing this one. However, one of the goals for this issue/PR was for the tests to complete the same tests before and after. This change replaces 1 test with 3 tests. 

The core suite now uses the onboarding tests from WC Admin, so you wouldn't see this in our E2E logs but the test log for this looks like:

```
  Store owner can go through store Onboarding
    ✓ can reset onboarding to default settings (86ms)
    ✓ can reset shipping zones to default settings (330ms)
    ✓ can reset to default settings (1364ms)
    ✓ can reset to default settings (1715ms)
    ✓ can reset to default settings (719ms)
    ✓ can reset shipping classes (239ms)
    ✓ can start and complete onboarding when visiting the site for the first time. (8739ms)
```

You can revert the changes to this file.

File: tests/e2e/core-tests/specs/keep/onboarding-tasklist.test.js [hunk 3]
```diff
@@ -33,11 +33,14 @@ const runOnboardingFlowTest = () => {
 				await withRestApi.deleteAllShippingZones();
 			});
 
-			it('can reset to default settings', async () => {
-				await withRestApi.resetSettingsGroupToDefault('general');
-				await withRestApi.resetSettingsGroupToDefault('products');
-				await withRestApi.resetSettingsGroupToDefault('tax');
-			});
+			it.each(['general', 'products', 'tax'])(
```
Comment (rrennick) [inline tests/e2e/core-tests/specs/keep/onboarding-tasklist.test.js]: Same feedback on changes to this file as previous comment.

**Review by rrennick — CHANGES_REQUESTED** [review]
@rodelgc Great work on this. Local testing looks like it may have reduced the runtime for the test suite by about 20 seconds.

In addition to the comments above, can you merge `trunk` again for the skip WCCOM test.

File: tests/e2e/core-tests/specs/keep/onboarding-tasklist.test.js (hunk 3 above)
Comment (rodelgc) [reply tests/e2e/core-tests/specs/keep/onboarding-tasklist.test.js]: Done reverting this one as well in https://github.com/woocommerce/woocommerce/commit/7a835abaead84ec806b9c00c85657c319be00dd7

File: tests/e2e/core-tests/specs/activate-and-setup/onboarding-tasklist.test.js (hunk 2 above)
Comment (rodelgc) [reply tests/e2e/core-tests/specs/activate-and-setup/onboarding-tasklist.test.js]: Thanks @rrennick ok cool. Makes sense. I've now reverted the each() call in https://github.com/woocommerce/woocommerce/pull/30119/commits/7a835abaead84ec806b9c00c85657c319be00dd7. I've also synced this branch with `trunk` as well.

**Review by rrennick — APPROVED** [review]
@rodelgc Thanks for the quick turn around on the review items.
